# HG changeset patch
# User Torsten <mttl@mailbox.org>
# Date 1551816411 -3600
#      Tue Mar 05 21:06:51 2019 +0100
# Branch stable
# Node ID b6dfe7d192bc83392465a890a8635cc81b17b8eb
# Parent  b00bd9afbed3c5c790e31d4d62b3ed1d4fd1a41b
fix creating keyword files for editor autocompletion (bug #55835)

* file-editor-tab.cc (update_lexer_settings): for testing whether qscintilla
  can open the files, try to open it readonly and if not existing, create
  path and try to open it writeonly

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -693,19 +693,26 @@ namespace octave
           }
         else  // no octave file, just add extension
           {
 
             _prep_apis_file = prep_apis_path + lexer->lexer () + ".pap";
 
           }
 
-        // make sure the apis file is usable
+        // Make sure the apis file is usable, otherwise the gui might crash,
+        // e.g., in case of max. number of opened files
         QFile f (_prep_apis_file);
-        bool apis_usable = f.open (QIODevice::ReadWrite);
+
+        bool apis_usable = f.open (QIODevice::ReadOnly);
+        if (! apis_usable)
+          {
+            QDir ().mkpath (QFileInfo (f).absolutePath ());
+            apis_usable = f.open (QIODevice::WriteOnly);
+          }
         if (apis_usable)
           f.close ();
 
         if (apis_usable &&
             (update_apis || ! _lexer_apis->loadPrepared (_prep_apis_file)))
           {
             // either we have decided to update the apis file or
             // no prepared info was loaded, prepare and save if possible
